### PR TITLE
locale.identifier is unreliable - use locale.preferred languages

### PR DESCRIPTION
**Pull request by mkaply**
*Thank you for submitting a pull request, your contributions are greatly appreciated!*

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.
## Pull Request Checklist

- [ ] My patch has gone through review and I have addressed review comments
- [ ] My patch has a standard commit message that looks like `Bug 12345678 - This fixes something something`

- [ ] I have updated the *Unit Tests* to cover new or changed functionality
- [ ] I have updated the *UI Tests* to cover new or changed functionality
- [ ] I have marked the bug with `[needsuplift]`
- [ ] I have made sure that localizable strings use `NSLocalizableString()`

## Screenshots

If your patch includes user interface changes that you would like to suggest or that you would like UX to look at, please include them here.

## Notes for testing this patch

If useful, please leave notes for QA, explaining what this patch changes and how it can be best tested and verified.

### CODE REVIEW

**Comment by mkaply** [comment]
@farhanpatel Need you to review. See https://bugzilla.mozilla.org/show_bug.cgi?id=1434195

File: Client/Frontend/Browser/SearchEngines.swift [hunk 1]
```diff
@@ -176,7 +176,7 @@ class SearchEngines {
     /// Get all bundled (not custom) search engines, with the default search engine first,
     /// but the others in no particular order.
     class func getUnorderedBundledEnginesFor(locale: Locale) -> [OpenSearchEngine] {
-        let languageIdentifier = locale.identifier
+        let languageIdentifier = Locale.preferredLanguages.first!
```
Comment (justindarc) [inline Client/Frontend/Browser/SearchEngines.swift]: `Locale.preferredLanguages.first ?? locale.identifier`

Comment (mkaply) [reply Client/Frontend/Browser/SearchEngines.swift]: I'm fine with that, but preferredLanguages should always have at least one value.

Comment (farhanpatel) [reply Client/Frontend/Browser/SearchEngines.swift]: `Locale.preferredLanguages.first ?? "en_US"` should work too

**Comment by farhanpatel** [comment]
@mkaply just gotta fix the forced unwrap and then we should be good to go!

**Pull request merged**